libr/core/cmd_anal.c

```diff
@@ -2170,8 +2170,8 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 		break;
 	case '-': // "af-"
 		if (!input[2] || !strcmp (input + 2, "*")) {
-			r_anal_fcn_del_locs (core->anal, UT64_MAX);
-			r_anal_fcn_del (core->anal, UT64_MAX);
+			r_list_purge (core->anal->fcns);
```

Comment:
what about having r_anal_fcn_del_all () instead of accessing internal fields from here?

Reply:
Do this when there is another command deleting all functions.

Reply:
This  also fixes an issue when you craft a function with `af+` which `r_anal_fcn_in` will return false.

Reply:
af-* deletes all functions

Reply:
just move this list_purge into that anal function

Reply:
> just move this list_purge into that anal function

?